File: RealEstateApi/Models/Booking.cs
using System;
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Mvc;


public class Booking
{   
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();  // Tạo UUID tự động khi khởi tạo Property

    [Required]
    public Guid  PropertyId { get; set; } // Khóa ngoại đến bất động sản

    [Required]
    public Guid  UserId { get; set; } // Khóa ngoại đến người dùng

    [Required]
    public DateTime BookingDate { get; set; } // Ngày đặt

    [Required]
    [DataType(DataType.Date)]
    public DateTime StartDate { get; set; } // Ngày bắt đầu thuê

    [Required]
    [DataType(DataType.Date)]
    public DateTime EndDate { get; set; } // Ngày kết thúc thuê

    [Required]
    [StringLength(20)]
    public string Status { get; set; } // Trạng thái đặt (confirmed/cancelled)
}

File: RealEstateFrontend/src/models/Booking.js
import { v4 as uuidv4 } from 'uuid';

export class Booking {
    constructor(id = uuidv4(), propertyId, userId, bookingDate, startDate, endDate, status) {
        this.id = id;
        this.propertyId = propertyId;
        this.userId = userId;
        this.bookingDate = bookingDate;
        this.startDate = startDate;
        this.endDate = endDate;
        this.status = status;
    }
}
